tensorboard/uploader/auth_test.py

```diff
@@ -149,5 +155,305 @@ def test_read_invalid_json_file(self):
             store.read_credentials()
 
 
+class FakeInstalledAppFlow:
+    """A minimal fake for the InstalledApp flow with the function we call.
+
+    This is a fake of a publicly available class that should already be tested,
+    so we mostly want to test high-level interactions with it.
+    """
+
+    def __init__(self, credentials=None, raiseError=False):
+        if not credentials and not raiseError:
+            ValueError("credentials cannot be None when raiseError is False")
+        if credentials and raiseError:
+            ValueError("credentials must be None when raiseError is True")
+        self.run_local_server_was_called = False
+        self._creds = credentials
+        self._raiseError = raiseError
+
+    def run_local_server(self, port=8080):
+        self.run_local_server_was_called = True
+        if self._raiseError:
+            raise webbrowser.Error()
+        return self._creds
+
+
+class AuthenticateUserTest(tb_test.TestCase):
+    def setUp(self):
+        super().setUp()
+        # Used to estimate if a browser is available in this env.
+        self.os_env_display_fn = self.enter_context(
+            mock.patch.object(os, "getenv")
+        )
+
+        self.mocked_installed_auth_flow_creator_fn = self.enter_context(
+            mock.patch.object(auth_flows.InstalledAppFlow, "from_client_config")
+        )
+
+        self.mocked_device_auth_flow = self.enter_context(
+            mock.patch.object(
+                auth, "_LimitedInputDeviceAuthFlow", autospec=True
+            )
+        )
+
+    def test_authenticate_user__no_force_console_override__has_display__uses_installed_app_flow(
```

Comment:
Trying to get these names to be more concise, I'd recommend just naming this:

test_uses_installed_app_flow(self):

* The test is already called AuthenticateUserTest, so perhaps "authenticate_user" is redundant.
* I also consider the "installed app flow" the happy path so it seems unnecessary to include all the scenarios where it might not be used in the name. It might just be preferable to include a comment in the test about the conditions in which its used.

Reply:
Done. Totally agree with the redundant authenticate_user.

For the rest... I think it's not always clear what the "happy path" is or why, and it could even change over time, so I like detailing the conditions for each test instead of assuming or describing one as the happy path. But I guess I agree it can be considerably less readable when trying to include the entire state in the test name, so I made them shorter.

Related to the verbosity of the state, I was trying to avoid using _when_ and _and_, to make them shorter, but with your suggestions I removed some of the state part, and I think I found a reasonable brief name that allows better readability.

Thanks for the review and all the helpful comments!